File: backend/GoIsland.Api/Models/Experience.cs
namespace GoIsland.Api.Models;

public class Experience
{
    public int Id { get; set; }
    public int HostId { get; set; }
    public string Slug { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string ShortDescription { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public int? DurationMinutes { get; set; }
    public string TimeZoneId { get; set; } = "America/Santo_Domingo";
    public string MeetingPointInstructions { get; set; } = string.Empty;
    public string? PickupInformation { get; set; }
    public string[] WhatIsIncluded { get; set; } = [];
    public string[] WhatIsNotIncluded { get; set; } = [];
    public string[] WhatToBring { get; set; } = [];
    public string GuestRequirements { get; set; } = string.Empty;
    public int? MinimumAge { get; set; }
    public string Difficulty { get; set; } = string.Empty;
    public string AccessibilityInformation { get; set; } = string.Empty;
    public string[] Languages { get; set; } = [];
    public string CancellationPolicy { get; set; } = string.Empty;
    public string[] Tags { get; set; } = [];
    public string Location { get; set; } = string.Empty;
    public decimal? Latitude { get; set; }
    public decimal? Longitude { get; set; }
    public string Category { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public int Capacity { get; set; }
    public int AvailableSpots { get; set; }
    public bool IsUnlimitedCapacity { get; set; }
    public bool IsApproved { get; set; }
    public string ApprovalStatus { get; set; } = ExperienceApprovalStatuses.Draft;
    public string? RejectionReason { get; set; }
    public DateTime? ReviewedAt { get; set; }
    public int? ReviewedByAdminId { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    public ICollection<ExperienceImage> Images { get; set; } = new List<ExperienceImage>();
    public ICollection<ExperienceItineraryItem> Itinerary { get; set; } = new List<ExperienceItineraryItem>();
}

File: frontend/src/types/experience.ts
export interface ExperienceImage {
  id: number;
  url: string;
  cardUrl: string;
  thumbnailUrl: string;
  altText: string;
  creditText: string;
  creditUrl: string | null;
  licenseName: string | null;
  licenseUrl: string | null;
  isCover: boolean;
  sortOrder: number;
}

export interface Experience {
  id: number;
  slug: string;
  title: string;
  shortDescription: string;
  description: string;
  durationMinutes: number | null;
  timeZoneId: string;
  meetingPointInstructions: string;
  pickupInformation: string | null;
  whatIsIncluded: string[];
  whatIsNotIncluded: string[];
  whatToBring: string[];
  guestRequirements: string;
  minimumAge: number | null;
  difficulty: string;
  accessibilityInformation: string;
  languages: string[];
  cancellationPolicy: string;
  tags: string[];
  itinerary: ExperienceItineraryItem[];
  location: string;
  latitude: number | null;
  longitude: number | null;
  distanceKm: number | null;
  category: string;
  price: number;
  capacity: number;
  availableSpots: number;
  isUnlimitedCapacity: boolean;
  schedulingMode?: string;
  images: ExperienceImage[];
  isApproved: boolean;
  createdAt: string;
  averageRating: number | null;
  reviewCount: number;
}

export interface ExperienceItineraryItem {
  id?: number;
  title: string;
  description: string;
  durationMinutes: number;
  location: string | null;
  sortOrder?: number;
}

export interface ExperienceSearchParams {
  query?: string;
  location?: string;
  category?: string;
  minPrice?: number;
  maxPrice?: number;
  from?: string;
  to?: string;
  quantity?: number;
  language?: string;
  difficulty?: string;
  accessible?: boolean;
  sort?: 'relevance' | 'newest' | 'priceAsc' | 'priceDesc' | 'rating';
  page?: number;
  pageSize?: number;
}

export interface PagedResponse<T> {
  items: T[];
  page: number;
  pageSize: number;
  totalItems: number;
  totalPages: number;
}
